tests/schema/tests.py

```diff
@@ -2758,6 +2758,53 @@ def test_alter_field_default_doesnt_perform_queries(self):
         with connection.schema_editor() as editor, self.assertNumQueries(0):
             editor.alter_field(AuthorWithDefaultHeight, old_field, new_field, strict=True)
 
+    @skipUnlessDBFeature('supports_foreign_keys')
+    def test_alter_field_fk_attributes_noop(self):
+        """
+        No queries are performed when changing FK field attributes that
+        don't affect the schema.
+        """
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+            editor.create_model(Book)
+        old_field = Book._meta.get_field('author')
+        new_field = ForeignKey(
+            Author, blank=True, editable=False, error_messages={'invalid': 'new error message'},
+            help_text='new help text', limit_choices_to={'foo': 'bar'}, related_name='new_related_name',
+            related_query_name='new_related_query_name', on_delete=PROTECT, validators=[lambda x: x],
+            verbose_name='new verbose name',
+        )
+        new_field.set_attributes_from_name('author')
```

Comment:
```suggestion
        new_field.set_attributes_from_name('renamed_author')
```
@charettes is this the change you are referring to? Tbh I only added this line because the previous test had it. I am not sure why we are calling `set_attributes_from_name` method here.